test/helpers/view_components/mentoring/inbox_test.rb

```diff
@@ -4,23 +4,21 @@ class MentoringInboxTest < ViewComponentTestCase
   test "mentoring inbox rendered correctly" do
     conversations_request = { endpoint: "conversations-endpoint" }
     tracks_request = { endpoint: "tracks-endpoint" }
-    sort_options = [
-      { value: 'recent', label: 'Sort by Most Recent' },
-      { value: 'exercise', label: 'Sort by Exercise' },
-      { value: 'student', label: 'Sort by Student' }
-    ]
 
-    assert_component_equal ViewComponents::Mentoring::Inbox.new(
-      conversations_request, tracks_request, sort_options
-    ).to_s,
-                           { id: "mentoring-inbox", props: {
-                             conversations_request: { endpoint: "conversations-endpoint" },
-                             tracks_request: { endpoint: "tracks-endpoint" },
-                             sort_options: [
-                               { value: 'recent', label: 'Sort by Most Recent' },
-                               { value: 'exercise', label: 'Sort by Exercise' },
-                               { value: 'student', label: 'Sort by Student' }
-                             ]
-                           } }
+    component = ViewComponents::Mentoring::Inbox.new(
+      conversations_request, tracks_request
+    ).to_s
+
+    assert_component_equal(
+      component,
+      {
+        id: "mentoring-inbox",
+        props: {
+          conversations_request: { endpoint: "conversations-endpoint" },
+          tracks_request: { endpoint: "tracks-endpoint" },
+          sort_options: ViewComponents::Mentoring::Inbox::SORT_OPTIONS
```

Comment:
I'm not a big fan of tests directly using values in the component being tested for the expected value. While I can see the advantage (changing it in the source still has the test passing), I feel that the test is now somehow less valuable, as any typos in the sort options would not get caught. It also limits refactoring of the sort options as there is this direct link. My personal preference would be to duplicate the sort options here and make it an internal details of the component.

Reply:
I agree with that. I'll update it.